### PR TITLE
[Bug] Prevent wild mons fleeing with U-turn, Flip Turn, Volt Switch

### PR DESCRIPTION
**Pull request by Fontbane**
## What are the changes the user will see?

- Fixes a bug or oversight where wild mons will flee after using U-turn, Flip Turn or Volt Switch.

## Why am I making these changes?

- In the main series games, these moves do not allow wild mons to flee. Teleport is the only self-switching move that allows fleeing.

## What are the changes from a developer perspective?

- Adds another check to the ForceSwitchOutAttr

### Screenshots/Videos

- Player using U-turn and switching, wild Weedle using U-turn and not fleeing, wild Rookidee using Teleport and fleeing
LoadSceneDemo7

## Checklist
- [X] **I'm using `beta` as my base branch**
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I considered writing automated tests for the issue?
- [ ] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -5502,6 +5502,12 @@ export class ForceSwitchOutAttr extends MoveEffectAttr {
       if (user.scene.currentBattle.waveIndex % 10 === 0) {
         return false;
       }
+
+      // Don't allow wild mons to flee with U-turn et al
+      if (this.selfSwitch && !user.isPlayer() && move.category !== MoveCategory.STATUS) {
+        return true;
```
Comment (DayKev) [inline src/data/move.ts:5508]:
```suggestion
        return false;
```
I think this should be `false`, otherwise seems good.

**Review by innerthunder — CHANGES_REQUESTED** [review]
We need a way to bypass this check for Teleport (maybe a `canFlee` field or similar?)

EDIT: Also, some moves (e.g. Shed Tail) need to outright fail under this condition. This still allows the move's other effects to go off.

**Comment by DayKev** [comment]
> We need a way to bypass this check for Teleport (maybe a `canFlee` field or similar?)

What do you mean? The video already shows Teleport still working after the change.

> EDIT: Also, some moves (e.g. Shed Tail) need to outright fail under this condition. This still allows the move's other effects to go off.

Ugh this attr needs a rewrite so badly...

**Comment by Snailman11** [comment]
Link to Innerthunder's comment (Discord Report), this could include Chilly Reception and Parting Shot

https://discord.com/channels/1125469663833370665/1289493136871264297

**Comment by PigeonBar** [comment]
Relevant issue: #4747

**Review by DayKev — DISMISSED** [review]

**Review by DayKev — APPROVED** [review]

**Review by MokaStitcher — APPROVED** [review]

**Review by innerthunder — APPROVED** [review]
lgtm (provided that Shed Tail and Baton Pass are made to fail in a future PR)

**Pull request merged**